docs/DeveloperGuide.md

```diff
@@ -2,14 +2,266 @@
+specific commands with specific options are more imperative, less readable, less maintainable and overall just a pain to
+handle. That's why the two classes are here.
+
+#### Alternative implementation
+
+No.
+
+### Review Flashcard Feature
+
+#### Current implementation
+
+The current review flashcard allows the user to review all the flashcards that are due today or before,
+it is implemented through the following steps:
+
+Step 1:
+The input of user is collected by `getUserCommand()` inside class `Ui`.
+
+Step 2:
+The input string will be converted into a `Command` object by being passed through
+`parseCommand(String userInput)` inside `Paser`.
+
+In this case, an `ReviewCommand` will be created and returned.
+
+Step 3:
+The `execute()` function of `ReviewCommand` will run, calling `getFlashCards()`
+of class `FlashcardList` to get the list of the flashcards.
+
+Then it will iterate through the `FlashcardList` and call the function `isDueBeforeToday()` of class `Flashcard` to
+check if the flashcard is due by today.
+
+If the flashcard is due by today, `reviewCurrentFlashcard(Ui display, Flashcard flashcard)` of class `ReviewCommand`
+will be called to review the card.
+
+First, the `Ui` will display the question of the current card by calling the `getQuestion()` method of
+class `Flashcard`, and ask user if user is ready to view the answer. After user enters any keyboard input, the answer of
+the current card will be shown by calling the `getAnswer()` method of class `Flashcard`, and `Ui` will ask the user if
+he/she has got the card correct. If the user inputs "y", then the current `Flashcard` is considered to be
+cleared and `updateDueDateAfterCorrectAnswer()` of `Flashcard` will be called to update its `dueDate`. Then Clanki will
+let user review the next `Flashcard`. If the user inputs "n", then the card is considered to be not cleared
+and `updateDueDateAfterIncorrectAnswer()` will be called to update its `dueDate`. Then Clanki will let user review the
+next `Flashcard`. This process will repeat until all the `Flashcards` in the `FlashcardList` are iterated.
+
+After the whole `FlashcardList` has been iterated through, a message congratulating the user that he/she has completed
+the reviewing task will be displayed.
+
+At this point, the reviewing process is completed and the program is ready to take another command.
+
+The following sequence diagram show how the review operation work:
+![ReviewFlashcard-0.png](umlDiagrams%2FReviewFlashcard-1.png)
+
+#### Reason for current implementation
+
+Through using `ReivewCommand` class, which extends `Command` class it increases the level of
+abstraction as the code can now perform the various commands on a class level.
+
+Moreover, `ReviewCommand` only has access to the public methods of `FlashcardList` and `Flashcard`, this reduces
+coupling in the program as the `ReviewCommand` will not have access to the inner structure of
+`FlashcardList` and `Flashcard`.
+
```

Comment:
You can consider adding a class diagram, at the start of the DG or otherwise, to show the overview of what is happening, to supplement the reader's understanding